server/src/cli.rs

```diff
@@ -340,19 +337,62 @@ pub struct InternalBackstageArgs {
     pub disable_instance_data_endpoint: bool,
 }
 
-#[derive(Args, Debug, Clone)]
-pub struct TokenHeader {
-    /// Token header to use for edge authorization.
-    #[clap(long, env, global = true, default_value = "Authorization")]
-    pub token_header: String,
+#[derive(Debug, Clone, Args)]
+pub struct AuthHeaders {
```

Comment:
I think it's unfortunate that we now propagate Option<String> throughout the system. This means doing an unwrap_or("Authorization") in several places in the heart of the system.

I'd suggest either mapping the CLI arg value to an internal non optional type like

```
struct InternalAuthHeaders { //yeah that name needs work
  //mirror fields but not optional
}
```

And then parsing to that type early in the startup sequence. That lets us push validation to the edge of the system (haha, see what I did there?)

Alternative a method on the struct itself that does that unwrap_or conversion would be a cheap way of getting something similar without future callers having to worry about resolving the Option.